iOS/DuckDuckGo/SettingsViewModel.swift

```diff
@@ -1048,14 +1048,22 @@ extension SettingsViewModel {
             let entitlementsToCheck: [Entitlement.ProductName] = [.networkProtection, .dataBrokerProtection, .identityTheftRestoration, .identityTheftRestorationGlobal, .paidAIChat]
 
             for entitlement in entitlementsToCheck {
-                if let hasEntitlement = try? await subscriptionAuthV1toV2Bridge.isEnabled(feature: entitlement),
+                if let hasEntitlement = try? await subscriptionAuthV1toV2Bridge.isFeatureEnabled(entitlement),
                     hasEntitlement {
                     currentEntitlements.append(entitlement)
                 }
             }
 
             self.state.subscription.entitlements = currentEntitlements
-            self.state.subscription.subscriptionFeatures = await subscriptionAuthV1toV2Bridge.currentSubscriptionFeatures()
+
+            // Note by Diego: using try? and then defaulting to [] below is not good!!!
+            //   I'm doing this because the original code did the same - it just did it inside
+            //   the call to currentSubscriptionFeatures so it wasn't as noticeable.  I'd like
+            //   to change this to have proper error handling but I see no quick solution
+            //   right now with the current code implementation.
+            //   In short: this is not new, I just surfaced it here instead of having it hidden
+            //   inside the call.
+            self.state.subscription.subscriptionFeatures = (try? await subscriptionAuthV1toV2Bridge.currentSubscriptionFeatures()) ?? []
```

Comment:
Should we consider allowing the cache to return the last known value even if expired - so we can avoid error handling in the UI level, which is tricky?  That'd default to showing the last known good state.

Reply:
One less bad option is defaulting to all features. This can be incorrect but less bad better than removing them.

Reply:
I've updated the comments to reference a follow-up work task.